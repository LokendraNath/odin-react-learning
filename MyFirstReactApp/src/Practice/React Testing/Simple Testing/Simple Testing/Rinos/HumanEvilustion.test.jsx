import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import HumanEvilustion from "./HumanEvilution";
import userEvent from "@testing-library/user-event";

describe("Human Revilution", () => {
  it("Should render Human", () => {
    const { container } = render(<HumanEvilustion />);
    expect(container).toMatchSnapshot();
  });

  it("should render human after button click", async () => {
    const user = userEvent.setup();

    render(<HumanEvilustion />);
    const button = screen.getByRole("button", { name: "Click Me" });

    await user.click(button);
    expect(screen.getByRole("heading").textContent).toMatch(/human/i);
  });
});
