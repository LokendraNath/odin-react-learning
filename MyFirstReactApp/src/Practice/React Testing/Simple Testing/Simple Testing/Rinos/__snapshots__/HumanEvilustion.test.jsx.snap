// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Human Revilution > Should render Human 1`] = `
<div>
  <button
    type="button"
  >
    Click Me
  </button>
  <h1>
    Monkey
  </h1>
</div>
`;

exports[`Human Revilution > Should render magnificent monkey 1`] = `
<div>
  <button
    type="button"
  >
    Click Me
  </button>
  <h1>
    Monkey
  </h1>
</div>
`;

exports[`Rinos Component > Should render magnificent monkey 1`] = `
<div>
  <button
    type="button"
  >
    Click Me
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;
